src/Header/ToggleAllTodos.tsx

```diff
@@ -0,0 +1,39 @@
+import classNames from 'classnames';
+import React from 'react';
+import { Todo } from '../types/todo';
+import { statusChanger } from '../utils/functions';
+
+type Props = {
+  todos: Todo[];
+  todoUpdater: (prevState: Todo[]) => void;
+};
+
+export const ToggleAllTodos: React.FC<Props> = ({ todos, todoUpdater }) => {
+  const uncompleted = todos.filter(todo => !todo.completed);
+
+  const handleAllCompleted = () => {
+    return uncompleted.length
+      ? todoUpdater(statusChanger(todos, false))
+      : todoUpdater(statusChanger(todos, true));
```

Comment:
What do you think, could you use smth like that instead?
```suggestion
    return todoUpdater(statusChanger(todos, !uncompleted.length));
```